### PR TITLE
[IMP] product: add specific security group for product models

### PR DESCRIPTION
**Pull request by ryv-odoo**
By default, an internal user cannot create products (expect when point_of_sale is installed, because this module makes, by default, all users pos manager). As we did for contacts, create a separate security group for product models.

task-4094988

### CODE REVIEW

**Review by Feyensv — COMMENTED** [review]
WIP or not WIP ? :D
(is there any task/ticket requesting internal users to be able to create products ?)

Also, wouldn't it be a good opportunity to simplify/remove existing rules by implying the product manager group on `sales_team.group_sale_manager` (and maybe some other groups) ?

**Comment by ryv-odoo** [comment]
Sry, I forget to push the new commit message. I added the task id.

> Also, wouldn't it be a good opportunity to simplify/remove existing rules by implying the product manager group on sales_team.group_sale_manager (and maybe some other groups) ?

Indeed, it sounds like a good idea. (`group_purchase_manager`, `group_sale_manager`; `stock.group_stock_manager` and `group_account_manager` are good candidates IMO)

File: addons/sales_team/security/sales_team_security.xml [hunk 1]
```diff
@@ -25,7 +25,8 @@
             <field name="category_id" ref="base.module_category_sales_sales"/>
             <field name="implied_ids" eval="[(4, ref('group_sale_salesman_all_leads')),
                          (4, ref('mail.group_mail_template_editor')),
-                         (4, ref('mail.group_mail_canned_response_admin'))]"/>
+                         (4, ref('mail.group_mail_canned_response_admin')),
+                         (4, ref('product.group_product_manager'))]"/>
```
Comment (Feyensv) [inline addons/sales_team/security/sales_team_security.xml:29]: `sales_team` doesn't depend on `product`, must be moved to `sale` :eyes:

**Comment by ryv-odoo** [comment]
I removed the last commit where some manager groups implied product manager. First, Product manager can be a specific role in an industry, then we don't want to be implied by others company roles. Also, the implies add to much right to some manager role on product, here is the list:

If we implied product manager to <group_manager>, people in that group will gain write/create/unlink for ...:
- mrp manager:
product_attribute
product_attribute_custom_value
product_attribute_value
product_category
product_combo
product_combo_item
product_template_attribute_exclusion
product_template_attribute_line
product_template_attribute_value

- pos manager:
product_packaging
product_pricelist_item
product_supplierinfo
product_tag

- purchase manager:
product_attribute
product_attribute_custom_value
product_attribute_value
product_tag

- purchase manager:
product_attribute
product_attribute_custom_value
product_attribute_value
product_combo
product_combo_item
product_pricelist
product_template_attribute_exclusion
product_template_attribute_line
product_template_attribute_value

- sale manager:
product_attribute_custom_value
product_combo
product_combo_item
product_packaging

- stock manager:
product_attribute_custom_value
product_combo
product_combo_item
  
cc @Feyensv

**Review by rco-odoo — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**